documents/functional_specifications/Functional Specification.md

```diff
@@ -0,0 +1,202 @@
+
+## 4. Design Constraints
+
+- As the app is intended for a very large population, its design should remain simple and intuitive. This will ensure an optimal user experience for both those who are very familiar with the technology and those who are less familiar with it.
+
+## 5. Assumptions
+
+- The application will assume users have access to a stable internet connection for full functionality.
+- Users will grant necessary permissions for the application to access location data, notifications, and other required services.
+- Users will provide accurate information during the account creation and profile setup process.
+
+## 6. Specific Function Descriptions
+
+### 6.1 Description
+
+- **Matching Function:** Matches candidates with job postings based on selected soft skills.
+- **Profile Management:** Allows users to create and manage their profiles, including updating personal information and soft skills.
+- **Job Posting Management:** Enables companies to create, edit, and manage job postings.
+
+### 6.2 Inputs
+
+- **Candidate Inputs:**
+  - Personal information,
+  - Selected soft skills,
+  - Job preferences.
+- **Company Inputs:**
+  - Job descriptions,
+  - Required soft skills,
+  - Location details.
+
+### 6.3 Processing
+
+- **Matching Algorithm:** Uses a weighted scoring system to match candidate soft skills with job requirements.
+- **Data Validation:** Ensures that user inputs meet predefined criteria for format, completeness, and accuracy.
+
+### 6.4 Outputs
+
+- **Match Results:**
+  - Displays matched job postings to candidates,
+  - Matched candidates to companies.
+- **Notifications:**
+  - Sends alerts to users when a company starts a chat,
+  - Important updates.
+
+## 7. User Interfaces
+
+The navigation flow of the windows, menus, and options is described, along with the expected content of each window. Examples of items included are screen resolutions, color scheme, primary font type and size. Discussion also includes how input validation will be done, and how data will be protected from accidental changes. Specific items are described for each screen such as input fields, control buttons, sizing options, and menus.
+
+You can find all the mockups here → [Mockups on Figma](https://www.figma.com/design/juw1xMWP0mUO0F4TyaW9jz/Flutter-Project?node-id=0%3A1&t=FrH8CO2qv7IewlBk-1)
+
+- First screen :
+  - Choose Your Language
+
+## 8. Hardware Interfaces
+
+- To use the "Adopt a Candidate" app, users looking for a job must access the app via their smartphone by downloading it from the Apple Store or Play Store. The app will be optimized to work on the following devices:
+  - Smartphones with iOS 12 or later.
+  - Smartphones with Android 8.0 (Oreo) or later.
+  - A stable Internet connection is required to access all the features of the application.
+- In a future release, users will also be able to use the app on their personal computers. The minimum specifications laid down are:
```

Comment:
When you use the term "users", I cannot understand if you're talking about jobseekers or companies. 
It's the case for all the times you used the word "user" when a feature is specific to a type of user.